project-api/ballerina-projects/src/main/java/io/ballerina/projects/balo/BaloProject.java

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.projects.balo;
+
+import io.ballerina.projects.PackageConfig;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.directory.PackageLoader;
+import io.ballerina.projects.env.BuildEnvContext;
+import io.ballerina.projects.environment.EnvironmentContext;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+/**
+ * {@code BaloProject} represents Ballerina project instance created from a balo.
+ *
+ * @since 2.0.0
+ */
+public class BaloProject extends Project {
+
+    /**
+     * Loads a BaloProject from the provided balo path.
+     *
+     * @param baloPath Balo path
+     * @return balo project
+     */
+    public static BaloProject loadProject(Path baloPath) {
+        Path absBaloPath = Optional.of(baloPath.toAbsolutePath()).get();
```

Comment:
Ideally we should do lazy loading without reading the ZIP. In a cached state what we want are the modules so we should so we should load only that. Shall we create an issue for that and merge this

Reply:
Added a issue https://github.com/ballerina-platform/ballerina-lang/issues/26196